superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -148,10 +172,10 @@ export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
   const handleOverwriteDataset = async () => {
     const [, key] = await Promise.all([
       updateDataset(
-        query.dbId,
+        datasource.dbId,
         datasetToOverwrite.datasetid,
-        query.sql,
-        query.results.selected_columns.map(
+        datasource.sql,
```

Comment:
so add chaining here as well, be consistent whenever we change with in the same path.

Reply:
add this back from @lyndsiWilliams PR:
```
if (!shouldOverwriteDataset) {
      setShouldOverwriteDataset(true);
      return;
}
``